core/Azure.Mcp.Core/src/Areas/Group/GroupSetup.cs

```diff
@@ -14,16 +13,16 @@ public sealed class GroupSetup : IAreaSetup
 
     public void ConfigureServices(IServiceCollection services)
     {
-        // No additional services needed for Group area
+        services.AddSingleton<GroupListCommand>();
     }
 
-    public void RegisterCommands(CommandGroup rootGroup, ILoggerFactory loggerFactory)
+    public CommandGroup RegisterCommands(IServiceProvider serviceProvider)
     {
-        // Create Group command group
         var group = new CommandGroup(Name, "Resource group operations - Commands for listing and managing Azure resource groups in your subscriptions.");
-        rootGroup.AddSubGroup(group);
 
         // Register Group commands
-        group.AddCommand("list", new GroupListCommand(loggerFactory.CreateLogger<GroupListCommand>()));
+        group.AddCommand("list", serviceProvider.GetRequiredService<GroupListCommand>());
```

Comment:
GroupListCommand has `public override string Name => "list";`
Wouldn't an extension method here simplify the call and get rid of the duplicate strings that have to stay in sinc?

```
group.AddCommand(serviceProvider.GetRequiredService<GroupListCommand>());

or

group.AddCommand<GroupListCommand>(serviceProvider);
```

Reply:
mmm Sure. That would be an easy fix.